enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,823 @@
+successful sync in the `IPAMClaim` status - or a failure otherwise.
+
+In this option, the [CNV to OVN-Kubernetes API](#cnv-ovnk-api) **remains** as
+defined today - i.e. OVN-Kubernetes will be pointed to an `IPAMClaim` via a
+particular annotation on the pod.
+
+The flow is described in more detail in the
+[De-centralized IP management workflow](#de-centralized-ip-management-workflow) section.
+
+##### De-centralized IP management workflow
+
+The [de-centralized IP management](#de-centralized-ip-management) flow is
+described in the following sequence diagram:
+
+```mermaid
+sequenceDiagram
+actor Admin
+actor Project admin
+
+participant MTV
+participant CNV
+participant o as OVN-Kubernetes
+
+Admin ->> o: cluster UDN
+o -->> Admin: OK
+
+Admin ->> CNV: provision IPAMClaim w/ MAC and IP requests
+CNV -->> Admin: OK
+
+Project admin ->> MTV: import VM
+MTV ->> CNV: create VM(name=<...>, primaryUDNMac=origMAC)
+CNV ->> CNV: ips = getIPsForMAC(mac=origMAC)
+CNV ->> o: create pod(mac=origMAC, IPs=ips)
+o -->> CNV: OK
+CNV -->> MTV: OK
+MTV -->> Project admin: OK
+```
+
+Hence, the required changes would be:
+- the `IPAMClaim` CRD status sub-resource will need to be updated, adding a
+  `Conditions` array. This way we will be able to report errors back to the
+  user (e.g. the desired IP allocation is already in use).
+- the `IPAMClaim` CRD spec will need to be updated, adding `MAC` and `IPs`
+  requests.
+- OVN-Kubernetes will need to try to apply the MAC and IP addresses requested
+  in its spec. If any conflicts are found for either MAC or IPs, the error will
+  be persisted in the `IPAMClaim` status (in the conditions array).
+- OVN-Kubernetes will record the requested IPs in the status sub-resource
+  (already does so today), along with a `Success` condition.
+
+This option has the network admin create the IPAMClaim, and request a MAC and
+IP addresses for the VM; in a second stage of the implementation, MTV (or any
+other component able to introspect the source cluster for the VM's MAC and IPs)
+will create the IPAMClaim on behalf of the admin user.
+
+###### Admin user flows for de-centralized approach
+
+The admin user will have to perform the following operations:
+- provision the cluster UDN CR having the overlay definition.
+  - layer2 topology ; IPAM lifecycle `Persistent` are a **must**
+- provision the `IPAMClaim` object, whose name **must** be created like
+  `<vm name>.<logical network name>`. The `IPAMClaim` must have both the IP and
+  MAC addresses for the VM attachment. This is only required until another tool
+  automatically provisions the `IPAMClaim`s on behalf of the admin
+
+As indicated above, when a tool exists that can introspect the source cluster
+to learn the VM's IP addresses, the admin user will only require provisioning
+the cluster UDN CR.
+
+###### Project admin flows for de-centralized approach
+
+The Project admin just has to use MTV to import the VM into CNV.
+
+## Test Plan
+
```

Comment:
test plan must include upgrade lane in downstream and ensuring after upgrade VM static config stays the same